src/sdc11073/mdib/consumermdib.py

```diff
@@ -340,294 +357,354 @@ def _update_from_mdib_version_group(self, mdib_version_group: MdibVersionGroupRe
-                                                'component states'):
-                    return
-                self._update_from_mdib_version_group(mdib_version_group)
-                states_by_handle = self.process_incoming_states_report(
-                    'component states', report, is_buffered_report)
+            if not self._can_accept_mdib_version(mdib_version_group.mdib_version, 'waveform states'):
+                return
+            self._update_from_mdib_version_group(mdib_version_group)
+            for state_container in state_containers:
+                old_state_container = self.states.descriptor_handle.get_one(state_container.DescriptorHandle,
+                                                                            allow_none=True)
+                if old_state_container is not None:
+                    if self._has_new_state_usable_state_version(old_state_container, state_container,
+                                                                'waveform states'):
+                        old_state_container.update_from_other_container(state_container)
+                        self.states.update_object(old_state_container)
+                        states_by_handle[old_state_container.DescriptorHandle] = old_state_container
+                else:
+                    self._logger.error('waveform states: got a new state {}',  # noqa: PLE1205
+                                       state_container.DescriptorHandle)
+                    self._set_descriptor_container_reference(state_container)
+                    self.states.add_object(state_container)
+                    states_by_handle[state_container.DescriptorHandle] = state_container
+
+            # add to Waveform Buffer
+            for state_container in states_by_handle.values():
+                state_container: RealTimeSampleArrayMetricStateContainer
+                descriptor_container = state_container.descriptor_container
+                d_handle = state_container.DescriptorHandle
+                rt_buffer = self.rt_buffers.get(d_handle)
+                if rt_buffer is None:
+                    sample_period = 0  # default
+                    if descriptor_container is not None:
+                        # read sample period
+                        sample_period = descriptor_container.SamplePeriod or 0
+                    rt_buffer = ConsumerRtBuffer(sample_period=sample_period,
+                                                 max_samples=self._max_realtime_samples)
+                    self.rt_buffers[d_handle] = rt_buffer
+                rt_sample_containers = rt_buffer.mk_rt_sample_containers(state_container)
+                rt_buffer.add_rt_sample_containers(rt_sample_containers)
         finally:
-            self.component_by_handle = states_by_handle  # used by wait_metric_matches method
+            self.waveform_by_handle = states_by_handle  # update observable
```

Comment:
because a description modification report is split into create, update and delete parts. These are optional, and the observable shall only be updated if the related report part is present.
There is no such optionality in waveform report.